File: frontend/src/services/productService.js
/**
 * Servicio para obtener productos.
 * Decide automáticamente si consumir desde tech_prod (API) o backend Django (DB).
 */

const PRODUCT_SOURCE = import.meta.env.VITE_PRODUCT_SOURCE || 'api';
const TECH_PROD_API_URL = import.meta.env.VITE_TECH_PROD_API_URL || 'http://localhost:5000';
const DJANGO_API_URL = import.meta.env.VITE_DJANGO_API_URL || 'http://localhost:8000';

// DEBUG: Ver configuración
console.log('🔧 PRODUCT_SOURCE:', PRODUCT_SOURCE);
console.log('🔧 TECH_PROD_API_URL:', TECH_PROD_API_URL);
console.log('🔧 DJANGO_API_URL:', DJANGO_API_URL);

/**
 * Obtiene la URL base según la configuración
 */
function getBaseUrl() {
  if (PRODUCT_SOURCE === 'api') {
    console.log('📡 Usando API directa (tech_prod)');
    return TECH_PROD_API_URL;
  }
  console.log('📡 Usando DB (Django backend)');
  return DJANGO_API_URL;
}

/**
 * Obtiene todos los productos
 * @returns {Promise<Array>} Lista de productos
 */
export async function getProducts() {
  const baseUrl = getBaseUrl();
  console.log('🌐 Fetching from:', `${baseUrl}/api/products`);
  
  const response = await fetch(`${baseUrl}/api/products`);
  
  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.error || 'Error al obtener productos');
  }
  
  const data = await response.json();
  // Si viene de Django, está en data.products
  // Si viene de tech_prod, puede estar en data.products o directamente data
  return data.products || data;
}

/**
 * Obtiene un producto por ID
 * @param {number|string} productId - ID del producto
 * @returns {Promise<Object>} Producto
 */
export async function getProductById(productId) {
  const baseUrl = getBaseUrl();
  const response = await fetch(`${baseUrl}/api/products/${productId}`);
  
  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.error || `Producto ${productId} no encontrado`);
  }
  
  const data = await response.json();
  return data.product || data;
}

/**
 * Verifica si la fuente es API (tech_prod directo)
 */
export function isApiSource() {
  return PRODUCT_SOURCE === 'api';
}

/**
 * Verifica si la fuente es DB (backend Django)
 */
export function isDbSource() {
  return PRODUCT_SOURCE === 'db';
}

File: backend/api/services/product_service.py
import os
import requests
from django.conf import settings


class ProductService:
    """
    Servicio para consumir productos desde tech_prod (API externa)
    y opcionalmente guardarlos en DB local.
    """
    
    def __init__(self):
        self.api_url = os.getenv('TECH_PROD_API_URL', 'http://localhost:5000')
        self.source = os.getenv('PRODUCT_SOURCE', 'api')
    
    def get_products(self):
        """
        Obtiene productos desde tech_prod API.
        Retorna lista de productos o None si hay error.
        """
        try:
            # Ruta correcta: /api/store/products
            response = requests.get(
                f"{self.api_url}/api/store/products",
                timeout=10
            )
            response.raise_for_status()
            return response.json()
        except requests.RequestException as e:
            print(f"Error fetching products from tech_prod: {e}")
            return None
    
    def get_product_by_id(self, product_id):
        """
        Obtiene un producto específico desde tech_prod API.
        """
        try:
            # Ruta correcta: /api/store/products/{id}
            response = requests.get(
                f"{self.api_url}/api/store/products/{product_id}",
                timeout=10
            )
            response.raise_for_status()
            return response.json()
        except requests.RequestException as e:
            print(f"Error fetching product {product_id}: {e}")
            return None
    
    def is_api_source(self):
        """Verifica si la fuente es API (frontend consume directo)"""
        return self.source == 'api'
    
    def is_db_source(self):
        """Verifica si la fuente es DB (backend hace proxy)"""
        return self.source == 'db'
